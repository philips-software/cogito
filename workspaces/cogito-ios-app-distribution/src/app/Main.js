import React from 'react'

import { PageCentered } from '@react-frontend-developer/react-layout-helpers'
import { ImageAppLink } from '../components/imageAppLink'
import { P, B } from 'glamorous'
import { Button } from 'semantic-ui-react'

class Main extends React.Component {
  render () {
    return (
      <PageCentered>
        <div style={{ width: '300px', height: '400px' }}>
          <P>This is the download page for <B>Cogito iOS App</B></P>
          <ImageAppLink />
          <Button
            basic
            color='pink'
            href='itms-services://?action=download-manifest&url=https://secure-transfer-app.charterlab.tech/manifest.plist'
            fluid
          >
            Download Cogito iOS App
          </Button>
        </div>
      </PageCentered>
    )
  }
}

export { Main }
