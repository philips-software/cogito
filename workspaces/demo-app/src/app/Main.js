import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { PageCentered } from '@react-frontend-developer/react-layout-helpers'
import { CogitoReact } from '@cogitojs/cogito-react'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import { Dimmer, Loader, Segment } from 'semantic-ui-react'
import { UserDataActions } from 'user-data'

import { CogitoId } from 'pages/home'
import { Contracts } from 'pages/contracts'
import { SimpleEncryption } from 'pages/simple-encryption'
import { StreamEncryption } from 'pages/stream-encryption'
import { Attestations } from 'pages/attestations'
import { NoMatch404 } from 'pages/404'

import { SimpleStorage } from '@cogitojs/demo-app-contracts'

const contractsInfo = {
  deployedContractsInfo: [
    { contractName: 'simpleStorage', contractDefinition: SimpleStorage }
  ],
  rawContractsInfo: []
}

class Main extends React.Component {
  renderComponent = (Component, routeProps, web3Props) => (
    <Component {...web3Props} {...routeProps} />
  )

  web3IsReady = ({ web3, channel, contracts }) => {
    return web3 && channel && contracts
  }

  onTelepathChanged = ({ channelId, channelKey, appName }, dispatch) => {
    dispatch(UserDataActions.setTelepath({ channelId, channelKey, appName }))
  }

  render () {
    return (
      <WithStore
        selector={state => ({
          channelId: state.userData.channelId,
          channelKey: state.userData.channelKey
        })}
        render={({ channelId, channelKey }, dispatch) => (
          <CogitoReact
            contracts={contractsInfo}
            channelId={channelId}
            channelKey={channelKey}
            appName='Cogito Demo App'
            onTelepathChanged={telepath =>
              this.onTelepathChanged(telepath, dispatch)}
          >
            {web3Props => {
              if (this.web3IsReady(web3Props)) {
                return (
                  <Switch>
                    <Route
                      exact
                      path='/'
                      render={routeProps =>
                        this.renderComponent(CogitoId, routeProps, web3Props)}
                    />
                    <Route
                      exact
                      path='/contracts'
                      render={routeProps =>
                        this.renderComponent(Contracts, routeProps, web3Props)}
                    />
                    <Route
                      exact
                      path='/simple-encryption'
                      render={routeProps =>
                        this.renderComponent(
                          SimpleEncryption,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route
                      exact
                      path='/stream-encryption'
                      render={routeProps =>
                        this.renderComponent(
                          StreamEncryption,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route
                      exact
                      path='/attestations'
                      render={routeProps =>
                        this.renderComponent(
                          Attestations,
                          routeProps,
                          web3Props
                        )}
                    />
                    <Route component={NoMatch404} />
                  </Switch>
                )
              } else {
                return (
                  <PageCentered>
                    <div style={{ width: '300px', height: '200px' }}>
                      <Segment style={{ width: '100%', height: '100%' }}>
                        <Dimmer active inverted>
                          <Loader inverted content={<p>loading web3...</p>} />
                        </Dimmer>
                      </Segment>
                    </div>
                  </PageCentered>
                )
              }
            }}
          </CogitoReact>
        )}
      />
    )
  }
}

export { Main }
