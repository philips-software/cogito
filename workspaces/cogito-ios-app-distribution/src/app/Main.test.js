import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { Main } from './Main'

it('renders without crashing', async () => {
  const div = document.createElement('div')
  await ReactDOM.render(<Main />, div)
})

it('renders correctly', async () => {
  const app = await renderer
    .create(<Main />)
  expect(app.toJSON()).toMatchSnapshot()
})
