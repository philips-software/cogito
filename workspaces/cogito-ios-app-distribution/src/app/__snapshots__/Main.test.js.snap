// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    style={
      Object {
        "height": "400px",
        "width": "300px",
      }
    }
  >
    <p
      className="emotion-0"
    >
      This is the download page for 
      <b
        className="emotion-0"
      >
        Cogito iOS App
      </b>
    </p>
    <a
      className="ui medium image"
      href="itms-services://?action=download-manifest&url=https://secure-transfer-app.charterlab.tech/manifest.plist"
      target="_blank"
    >
      <img
        src="test-file-stub"
      />
    </a>
    <a
      className="ui pink basic fluid button"
      href="itms-services://?action=download-manifest&url=https://secure-transfer-app.charterlab.tech/manifest.plist"
      onClick={[Function]}
      role="button"
    >
      Download Cogito iOS App
    </a>
  </div>
</div>
`;
